### PR TITLE
Camera features - Flash, My QR Code and Browse

### LINKED ISSUE #1495: Update Light, My QR Code, Browse buttons in Scan QR Code screen
**Current:**

**We need:** 
a) Light - tour on the light in your camera
b) My QR Code - switches to "My Wallet Address" screen
c) Browse - to select images from your Camera Roll

Screenshot 2020-07-03 10 36 30

### PR DESCRIPTION
**Pull request by ChintanRathod**
Fixes #1495

### CODE REVIEW

**Comment by ChintanRathod** [comment]
@JamesSmartCell This PR has all 3 buttons implementation. But there is an issue I am facing with Zxing library to decode bitmap we select from "Browse". I tried to fix an issue of Zxing library decode but not able to fix it. With this issue, after selecting image, we are getting "com.google.zxing.NotFoundException". I will check again to fix this issue.

**Comment by JamesSmartCell** [comment]
> @JamesSmartCell This PR has all 3 buttons implementation. But there is an issue I am facing with Zxing library to decode bitmap we select from "Browse". I tried to fix an issue of Zxing library decode but not able to fix it. With this issue, after selecting image, we are getting "com.google.zxing.NotFoundException". I will check again to fix this issue.

I expected that would be an issue. I wasn't sure if the decode library would accept an image directly or not.

**Comment by ChintanRathod** [comment]
I tried with different solutions like Resize, rotate and scaling. Yet not able to find the root cause.  
2 observations, 
- Screenshots are decoded successfully
- Camera pictures having exception

**Comment by JamesSmartCell** [comment]
Can we lock the orientation while in this mode so we don't need to worry about it?

**Comment by hboon** [comment]
@JamesSmartCell QR code from photo library: if it's tedious/buggy, might be worth just skipping the feature? It doesn't seem like an incredibly useful feature. I only use it because I normally develop for iOS in the simulator (which doesn't have a camera).

**Comment by ChintanRathod** [comment]
I went through many Stackoverflow and blogs which states that Zxing library is built for scanning through preview and has no core functionality to decode captured image. Many people are facing same issue yet no solid solution I was able to find. 
Ref. 
https://github.com/journeyapps/zxing-android-embedded/issues/395

**Comment by JamesSmartCell** [comment]
Removing the browse seems to be the best solution. Is the 'plumbing' for the browse still present or has it been removed too?

File: app/src/main/java/com/alphawallet/app/ui/HomeActivity.java [hunk 1]
```diff
@@ -945,8 +945,20 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data)
             case C.REQUEST_UNIVERSAL_SCAN:
                 if(resultCode == Activity.RESULT_OK)
                 {
-                    String qrCode = data.getStringExtra(C.EXTRA_QR_CODE);
-                    viewModel.handleQRCode(this, qrCode);
+                    if(data.hasExtra(C.EXTRA_QR_CODE))
```
Comment (JamesSmartCell) [inline app/src/main/java/com/alphawallet/app/ui/HomeActivity.java:948]: Please use a space between if and the bracket

File: app/src/main/java/com/alphawallet/app/ui/QRScanning/CameraPreview.java [hunk 2]
```diff
@@ -309,4 +309,31 @@ public void onAutoFocus(boolean success, Camera camera) {
     private void scheduleAutoFocus() {
         mAutoFocusHandler.postDelayed(doAutoFocus, 1000);
     }
+
+    public void turnOffFlash() {
+        try {
```
Comment (JamesSmartCell) [inline app/src/main/java/com/alphawallet/app/ui/QRScanning/CameraPreview.java:314]: inconsistent bracing - the preferred code standard for AW is to use a separate line for bracing eg:
```
if (a=b)
{
    do something;
}
```

File: app/src/main/java/com/alphawallet/app/ui/zxing/QRScanningActivity.java [hunk 3]
```diff
@@ -118,4 +189,71 @@ public void onBackPressed()
...
+                        .subscribe(this::onSuccess, this::onError);
+            }
+        }
+    }
+
+    private void onError(Throwable throwable)
+    {
+        Toast.makeText(this, getString(R.string.error_browse_selection), Toast.LENGTH_LONG).show();
+    }
+
```
Comment (JamesSmartCell) [inline app/src/main/java/com/alphawallet/app/ui/zxing/QRScanningActivity.java:214]: The error should be handled by the calling function - usually a dialog box.

**Comment by JamesSmartCell** [comment]
@ChintanRathod just a few formatting changes for some bad formatting that's crept in.

When handling the errors - can you check if these are handled by the activity/fragment that's calling this function? In most cases it should be handled there. So - if something is scanned and the QR code can't be classified and flow is passed back to the HomeActivity then the handling should be in there.

**Comment by ChintanRathod** [comment]
@JamesSmartCell Understood.

**Pull request merged**